frontend/src/composables/useEditorLineHighlighter.js

```diff
@@ -9,94 +9,154 @@ import {
+import {
+  EditorView,
+  lineNumbers,
+  Decoration,
+} from '@codemirror/view'
+import {
+  StateEffect,
+  EditorSelection,
+  StateField,
+} from '@codemirror/state'
+
+export function useEditorLineHighlighter (cmView) {
+  cmView.dom.classList.add('g-cm-line-highlighter')
+
+  const highlightableLineNumbers = lineNumbers({
+    domEventHandlers: {
+      mousedown (cmView, line, event) {
+        onGutterClick(cmView, line, event)
+      },
+    },
+  })
 
-export function useEditorLineHighlighter (cm) {
   const params = useUrlSearchParams('hash-params')
 
+  const valuesFromParams = () => {
+    const line = params.line ?? ''
+    const values = line.split('-')
+      .map(value => parseInt(value, 10))
+      .sort((a, b) => a - b)
+    return values.every(Number.isInteger)
+      ? values
+      : []
+  }
+
   const selection = computed({
     get () {
-      const line = params.line ?? ''
-      const values = line.split('-')
-        .map(value => parseInt(value, 10) - 1)
-        .sort((a, b) => a - b)
-      return values.every(Number.isInteger)
-        ? values
-        : []
+      return valuesFromParams()
     },
     set (values = []) {
-      const [startLine, endLine = startLine] = values
+      const valuesCurrent = valuesFromParams()
+
+      let [startLine, endLine = startLine] = values
+      const [startLineCurrent, endLineCurrent] = valuesCurrent
+
+      if (endLine === startLine && startLine === startLineCurrent && !endLineCurrent) {
+        startLine = null
+      }
+
       if (!startLine || !values.every(Number.isInteger)) {
         params.line = null
       } else if (startLine === endLine) {
-        params.line = `${startLine + 1}`
+        params.line = `${startLine}`
       } else if (startLine > endLine) {
-        params.line = `${endLine + 1}-${startLine + 1}`
+        params.line = `${endLine}-${startLine}`
       } else {
-        params.line = `${startLine + 1}-${endLine + 1}`
+        params.line = `${startLine}-${endLine}`
       }
     },
   })
 
   const startLine = computed(() => selection.value[0])
 
   if (startLine.value) {
-    cm.scrollIntoView({ line: startLine.value, ch: 0 })
-    cm.setCursor({ line: startLine.value, ch: 0 })
+    const pos = cmView.state.doc.line(startLine.value).from
+
+    cmView.dispatch({
+      selection: EditorSelection.cursor(pos),
+      effects: EditorView.scrollIntoView(pos, { y: 'center' }),
+    })
   }
 
+  function onGutterClick (cmView, line, event) {
+    const lineNumber = cmView.state.doc.lineAt(line.from).number
+    selection.value = startLine.value && startLine.value !== lineNumber && event.shiftKey
+      ? [startLine.value, lineNumber]
+      : [lineNumber]
+  }
+
+  const setHighlightEffect = StateEffect.define()
+  const highlightField = StateField.define({
+    create () {
+      return Decoration.none
+    },
+    update (highlights, tr) {
+      // need to re-add when editor updates
+      cmView.dom.classList.add('g-cm-line-highlighter')
```

Comment:
This is because of the border in the g-highlighted css classes. Not sure why this did not affect cm5 lines... However, I changed the css to use box-shadow for the left border and pseudo-elements for top and bottom lines (box-shadow did not work here as the classes would overwrite each-other)